src/bootstrap/src/core/builder/tests.rs

```diff
@@ -1606,12 +1606,8 @@ mod snapshot {
             ctx.config("test")
                 .path("cargo")
                 .render_steps(), @r"
-        [build] llvm <host>
-        [build] rustc 0 <host> -> rustc 1 <host>
-        [build] rustc 1 <host> -> std 1 <host>
-        [build] rustc 1 <host> -> rustc 2 <host>
-        [build] rustc 2 <host> -> cargo 3 <host>
-        [build] rustdoc 2 <host>
+        [build] rustc 0 <host> -> cargo 1 <host>
+        [build] rustdoc 0 <host>
```

Comment:
Question [CARGO_TEST_STAGING 1/2]: there's some funky staging going on with the staging, I think? Usually `./x test tests/ui --stage=1` means to test xxx *using* the stage 1 (compiler, library). However, `./x test cargo` here means... `--stage=0`?

Reply:
Ok that's funny. I inadvertedly made testing stage 0 rustc with cargo work =D As the code says, cargo now tests the compiler that *builds* Cargo. So when we build stage 1 cargo, we then test stage 0 rustc.. that's not ideal, hmm. Ideally we build cargo with stage 0 rustc and then build stage 1 rustc and test that, but that causes the sysroot issues. I think I'll just modify the cargo test step to build cargo stage N using rustc stage N and then make it test rustc stage N...